### PR TITLE
[e2e tests]: Ignore resourceVersion too old watch error

### PR DESCRIPTION
**Pull request by akalenyu**
Signed-off-by: Alex Kalenyuk <akalenyu@redhat.com>

**What this PR does / why we need it**:
A couple of related failures:
https://prow.ci.kubevirt.io/view/gs/kubevirt-prow/logs/periodic-kubevirt-e2e-k8s-1.26-centos9-sig-storage/1620975484743454720
https://prow.ci.kubevirt.io/view/gs/kubevirt-prow/logs/periodic-kubevirt-e2e-k8s-1.26-centos9-sig-storage/1622304257438912512
Readouts:
https://sdk.operatorframework.io/docs/faqs/#i-see-the-warning-in-my-operators-logs-the-resourceversion-for-the-provided-watch-is-too-old-whats-wrong
https://github.com/kubernetes/kubernetes/issues/72187

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by xpivarc** [comment]
Can you include ref in PR description, please?

**Comment by akalenyu** [comment]
> Can you include ref in PR description, please?

Yep, done

**Review by fossedihelm — APPROVED** [review]

**Comment by fabiand** [comment]
Thanks, great improvement (seriously)

Bring distance between signal and noise.

**Comment by akalenyu** [comment]
> /hold
> 
> ```
> "kind": "DataVolume",
>             "apiVersion": "cdi.kubevirt.io/v1beta1",
>             "metadata": {
>                 "name": "test-datavolume-6gs9qdltq7pb",
>                 "namespace": "kubevirt-test-default1",
>                 "uid": "43598634-d17b-401c-b4c0-e8ccec2748e4",
>                 "resourceVersion": "61615",
>                 "generation": 250,
> ```
> 
> Note the generation field, we should probably pursue that instead of sweeping the err https://prow.ci.kubevirt.io/view/gs/kubevirt-prow/pr-logs/pull/kubevirt_kubevirt/9134/pull-kubevirt-e2e-k8s-1.26-centos9-sig-storage/1622969537487966208

/hold cancel
this is behaving as intended, the DV is reporting progress every couple seconds so really no way to avoid that

**Comment by akalenyu** [comment]
ping @xpivarc @fossedihelm

**Comment by fossedihelm** [comment]
@xpivarc Can you have a look? 1.26 storage lane is affected by this error several times, and this could help stabilizing the lane. Thanks!

File: tests/watcher/watcher.go [hunk 1]
```diff
@@ -184,9 +184,7 @@ func (w *ObjectEventWatcher) Watch(ctx context.Context, processFunc ProcessFunc,
 				switch watchEvent.Object.(type) {
 				case *metav1.Status:
 					status := watchEvent.Object.(*metav1.Status)
-					//api server sometimes closes connections to Watch() client command
-					//ignore this error, because it will reconnect automatically
-					if status.Message != "an error on the server (\"unable to decode an event from the watch stream: http2: response body closed\") has prevented the request from succeeding" {
+					if !shouldIgnoreWatchErr(status) {
```
Comment (xpivarc) [inline tests/watcher/watcher.go:187]: Do we handle reconnects?

Comment (akalenyu) [reply tests/watcher/watcher.go:187]: yeah we had this prior to this PR, ofc I kept it in `shouldIgnoreWatchErr`

**Comment by acardace** [comment]
/approve
very nice! Thanks @akalenyu

**Review by xpivarc — COMMENTED** [review]
/hold
Don't you need to do another watch?

**Comment by akalenyu** [comment]
> /hold Don't you need to do another watch?

which one? I don't think so

File: tests/watcher/watcher.go [hunk 2]
```diff
@@ -274,3 +272,18 @@ const (
...
 	watchSinceResourceVersion startType = "watchSinceResourceVersion"
 )
+
+func shouldIgnoreWatchErr(status *metav1.Status) bool {
+	//api server sometimes closes connections to Watch() client command
+	//ignore this error, because it will reconnect automatically
+	if status.Message == "an error on the server (\"unable to decode an event from the watch stream: http2: response body closed\") has prevented the request from succeeding" {
+		return true
+	}
+	// https://sdk.operatorframework.io/docs/faqs/#i-see-the-warning-in-my-operators-logs-the-resourceversion-for-the-provided-watch-is-too-old-whats-wrong
```
Comment (rmohr) [inline tests/watcher/watcher.go:282]: I can be wrong but I am pretty certain that the reconnect is done by the informer like this: If connection fails, it lists again and then creates a new watch with the desired resource version as a start. When we swallow this error, based on the usage of the watch, we either miss a warning which we want to let the test fail, or we may miss the actual desired state update and stall the test until it hits a timeout.

Again, I am not 100% sure but almost that we need to re-establish the watch when we try to handle this error.

Comment (xpivarc) [reply tests/watcher/watcher.go:282]: Ref for the case you mention is here https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/client-go/rest/request.go#L714 but it does not cover old rv

**Comment by akalenyu** [comment]
/reopen
we still want to pursue this https://github.com/kubevirt/kubevirt/pull/9752#pullrequestreview-1435530041

**Comment by xpivarc** [comment]
@akalenyu Can we close this?

**Comment by akalenyu** [comment]
> @akalenyu Can we close this?

Yep, the reference https://github.com/kubevirt/kubevirt/pull/9175#discussion_r1151830549 will stay
here in case we ever need to tackle this error
/close

**Pull request closed without merging**